File: docs/snippets/all/tutorials/any_values.rs
//! Log arbitrary data.

use std::sync::Arc;

use rerun::external::arrow;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let rec = rerun::RecordingStreamBuilder::new("rerun_example_any_values").spawn()?;

    let any_values = rerun::AnyValues::default()
        // Using arbitrary Arrow data.
        .with_component_from_data(
            "homepage",
            Arc::new(arrow::array::StringArray::from(vec![
                "https://www.rerun.io",
            ])),
        )
        .with_component_from_data(
            "repository",
            Arc::new(arrow::array::StringArray::from(vec![
                "https://github.com/rerun-io/rerun",
            ])),
        )
        // Using Rerun's builtin components.
        .with_component::<rerun::components::Scalar>("confidence", [1.2, 3.4, 5.6])
        .with_component::<rerun::components::Text>("description", vec!["Bla bla bla…"]);

    rec.log("any_values", &any_values)?;

    Ok(())
}

File: docs/snippets/all/tutorials/any_values.py
"""Log arbitrary data."""

import rerun as rr

rr.init("rerun_example_any_values", spawn=True)

rr.log(
    "any_values",
    rr.AnyValues(
        # Using arbitrary Arrow data.
        homepage="https://www.rerun.io",
        repository="https://github.com/rerun-io/rerun",
    )
    # Using Rerun's builtin components.
    .with_component("confidence", rr.components.ScalarBatch._COMPONENT_TYPE, [1.2, 3.4, 5.6])
    .with_component("description", rr.components.TextBatch._COMPONENT_TYPE, "Bla bla bla…"),
)
